Muart-C/politico
# #163900813 add register candidate endpoint

**What does this PR do?**
 - Add a register a candidate endpoint.

**Description of Task to be completed?**
- Add functionality to check if the user exists before registration.
- Add functionality to check if a party exists before registering a candidate.
- Add functionality to check if an office exists before registering a candidate.


**How should the app be tested?**
-  `git clone https://github.com/Muart-C/politico.git`

- `cd politico`

- `git checkout ft-register-candidate-163900813`

-  `$ export FLASK_APP="run.py"`
   `$ export FLASK_ENV="development"`
   `$ export FLASK_DEBUG=1`
   `$ export DATABASE_POLITICO="politico"`
   `$ export DATABASE_USERNAME="yourpostgresusername"`
   `$ export DATABASE_PASSWORD="yourpostgrespassword"`
   `$ export DATABASE_HOST="localhost"`
   `$ export DATABASE_PORT=5432`
 - Or alternatively you can create a .env file and add the above configurations on your local machine.

-  Then start your app via the command `flask run` 

- First of all, ensure you install Postman - Test client to test the API endpoint.

[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/4e6c88cc481fc0ceefd0) via your web browser.


**Steps to follow on postman**

- Send POST requests to create a user, a party and an office to create a party.
- A candidate should be registered and a 201 response status returned.
 else a 404 error will be returned if either the party, office and the user was not registered

**What are the relevant pivotal tracker stories**
https://www.pivotaltracker.com/story/show/163900813

**screenshot**
![screenshot from 2019-02-19 10-57-29](https://user-images.githubusercontent.com/28337458/52998853-570ea780-3435-11e9-92ff-ef7180a44ecd.png)
